### PR TITLE
Update tspconfig.yaml

### PR DESCRIPTION
**Pull request by raych1**
The purpose is to have each language emitter to use the same config value in this file for code generation. Language emitters have the knowledge of the code generation path under sdk project directory base on the sdk project template.
 - clean up parameters section
 - get rid of `emitter-out-dir`
 - introduce `service-dir` and `package-dir` to be used by emitter and tools to compute the sdk project directory
 - add `additionalDirectories` to parameters section

### CODE REVIEW

**Comment by weshaggard** [comment]
I like this change but lets please try to get the sdk automation working before merging.

**Review by ihsanali01 — APPROVED** [review]
Approved

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml [hunk 1]
```diff
@@ -17,22 +13,22 @@ options:
     emitter-output-dir: "{project-root}/.."
     examples-directory: "./examples"
     output-file: "{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/widgets.json"
-  "@azure-tools/typespec-python":
-    emitter-output-dir: "{python-sdk-folder}/sdk/{service-directory-name}/{package-name}"
+  "@azure-tools/typespec-python":    
+    package-dir: "azure-contoso-widgetmanager"
     package-mode: "dataplane"
-    package-name: "azure-contoso-widgetmanager"
+    package-name: "{package-dir}"
```
Comment (weshaggard) [inline specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:19]: @lmazuel can we have this package-name property defaulted to {package-dir} so we don't have to specify it in every config. Similar do we need to set the package-mode?

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml [hunk 2]
```diff
@@ -17,22 +13,22 @@ options:
...
     package-mode: "dataplane"
-    package-name: "azure-contoso-widgetmanager"
+    package-name: "{package-dir}"    
   "@azure-tools/typespec-csharp":
+    package-dir: "Azure.Contoso.WidgetManager"
     clear-output-folder: true
-    emitter-output-dir: "{csharp-sdk-folder}/sdk/{service-directory-name}/{namespace}/src"
     model-namespace: false
-    namespace: Azure.Contoso.WidgetManager
+    namespace: "{package-dir}"
```
Comment (weshaggard) [inline specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:24]: @m-nash can we default the namespace to package-dir to eliminate this needed in the config? Also can we eliminate some of the other options with nices defaults?

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml [hunk 3]
```diff
@@ -17,22 +13,22 @@ options:
...
-    emitter-output-dir: "{csharp-sdk-folder}/sdk/{service-directory-name}/{namespace}/src"
     model-namespace: false
-    namespace: Azure.Contoso.WidgetManager
+    namespace: "{package-dir}"    
   "@azure-tools/typespec-ts":
-    emitter-output-dir: "{js-sdk-folder}/sdk/{service-directory-name}/contosowidgetmanager-rest"
+    package-dir: "contosowidgetmanager-rest"
     generateMetadata: true
     generateTest: true
     packageDetails:
```
Comment (weshaggard) [inline specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:29]: @joheredi @MaryGao did we figure out if we can eliminate some of these configurations with smart defaults?

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml (hunk 2 above)
Comment (m-nash) [reply specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:24]: https://github.com/Azure/autorest.csharp/issues/3408

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml [hunk 4]
```diff
@@ -17,22 +13,22 @@ options:
     emitter-output-dir: "{project-root}/.."
     examples-directory: "./examples"
     output-file: "{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/widgets.json"
-  "@azure-tools/typespec-python":
-    emitter-output-dir: "{python-sdk-folder}/sdk/{service-directory-name}/{package-name}"
+  "@azure-tools/typespec-python":
```
Comment (weidongxu-microsoft) [inline specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:16]: nit, clean the extra spaces

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by msyyc — APPROVED** [review]

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml (hunk 3 above)
Comment (MaryGao) [reply specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:29]: I think we could remove below config:

```
generateMetadata: true
generateTest: true
packageDetails:
    description: "Contoso WidgetManager Service"
```

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml (hunk 1 above)
Comment (raych1) [reply specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:19]: @msyyc approved this change.

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml (hunk 3 above)
Comment (raych1) [reply specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml:27]: Removed.

File: specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml (hunk 4 above)
Comment (raych1) [reply specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml]: Removed.

**Comment by raych1** [comment]
dotnet sdk generation fails due to known issue of project scaffolding.

**Pull request merged**